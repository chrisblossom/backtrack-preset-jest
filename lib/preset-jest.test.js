'use strict';

const transformConfig = require('@backtrack/core/dist/options-file/transform-config')
    .transformConfig;

test('preset returns expected config - defaults', () => {
    const backtrackConfig = {
        presets: ['../'],
    };

    const result = transformConfig(backtrackConfig, __dirname);

    expect(result).toMatchSnapshot();
});

test('preset returns expected config - windows = false', () => {
    const backtrackConfig = {
        presets: [['../', { windows: false }]],
    };

    const result = transformConfig(backtrackConfig, __dirname);

    expect(result).toMatchSnapshot();
});

test('preset returns expected config - isApp = true', () => {
    const backtrackConfig = {
        presets: [['../', { isApp: true }]],
    };

    const result = transformConfig(backtrackConfig, __dirname);

    expect(result).toMatchSnapshot();
});

test('preset returns expected config - isApp = true, windows false', () => {
    const backtrackConfig = {
        presets: [['../', { isApp: true, windows: false }]],
    };

    const result = transformConfig(backtrackConfig, __dirname);

    expect(result).toMatchSnapshot();
});
