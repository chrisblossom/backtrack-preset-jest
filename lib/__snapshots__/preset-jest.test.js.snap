// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preset returns expected config - defaults 1`] = `
Object {
  "files": Object {
    "dest": Object {
      "absolute": Object {
        ".circleci/config.yml": "<PROJECT_ROOT>/.circleci/config.yml",
        "appveyor.yml": "<PROJECT_ROOT>/appveyor.yml",
        "jest.config.js": "<PROJECT_ROOT>/jest.config.js",
        "wallaby.config.js": "<PROJECT_ROOT>/wallaby.config.js",
      },
      "allowChanges": Object {
        ".circleci/config.yml": false,
        "appveyor.yml": false,
        "jest.config.js": false,
        "wallaby.config.js": false,
      },
      "files": Array [
        "wallaby.config.js",
        ".circleci/config.yml",
        "jest.config.js",
        "appveyor.yml",
      ],
      "ignoreUpdates": Object {
        ".circleci/config.yml": false,
        "appveyor.yml": false,
        "jest.config.js": false,
        "wallaby.config.js": false,
      },
    },
    "makeDirs": Array [
      ".circleci",
      "dist",
      "src",
    ],
    "src": Object {
      "absolute": Object {
        "lib/files/appveyor.yml": "<PROJECT_ROOT>/lib/files/appveyor.yml",
        "lib/files/circleci-v2.yml": "<PROJECT_ROOT>/lib/files/circleci-v2.yml",
        "lib/files/jest-library.js": "<PROJECT_ROOT>/lib/files/jest-library.js",
        "lib/files/wallaby.js": "<PROJECT_ROOT>/lib/files/wallaby.js",
      },
      "files": Array [
        "lib/files/wallaby.js",
        "lib/files/circleci-v2.yml",
        "lib/files/jest-library.js",
        "lib/files/appveyor.yml",
      ],
      "hash": Object {
        "lib/files/appveyor.yml": "e26fc69a9c740e6fe6267ea2da435aa54c83a110eface66642470dc2e8336713",
        "lib/files/circleci-v2.yml": "3b7696d7b955ebd4ac0f5eeacfff227acea15301b6e450b111677d32411b2a8b",
        "lib/files/jest-library.js": "322dd136d43514064c38fdf8cd999d6511d83af8efbb35d8d355a3f35937c6c4",
        "lib/files/wallaby.js": "bfd3a73af1c8ecd93d812d70e4e599406f14eda36920955df69a4109a3a2ed2e",
      },
    },
  },
  "packageJson": Array [
    Object {
      "scripts": Object {
        "test.update": "jest --updateSnapshot",
        "test.watch": "jest --watch",
      },
    },
  ],
  "prepublishOnly": Array [
    "backtrack test",
  ],
  "prepush": Array [
    "backtrack test",
  ],
  "resolve": Object {
    "@backtrack/preset-jest": "<PROJECT_ROOT>/lib",
  },
  "test": Array [
    "jest --passWithNoTests",
  ],
  "test.ci": Array [
    "jest --passWithNoTests --coverage --maxWorkers 2",
    Object {
      "name": "codecov",
      "task": "codecov",
    },
  ],
  "test.ci-pretest": Array [
    [Function noop],
  ],
}
`;

exports[`preset returns expected config - isApp = true 1`] = `
Object {
  "files": Object {
    "dest": Object {
      "absolute": Object {
        ".circleci/config.yml": "<PROJECT_ROOT>/.circleci/config.yml",
        "appveyor.yml": "<PROJECT_ROOT>/appveyor.yml",
        "jest.config.js": "<PROJECT_ROOT>/jest.config.js",
        "wallaby.config.js": "<PROJECT_ROOT>/wallaby.config.js",
      },
      "allowChanges": Object {
        ".circleci/config.yml": false,
        "appveyor.yml": false,
        "jest.config.js": false,
        "wallaby.config.js": false,
      },
      "files": Array [
        "wallaby.config.js",
        ".circleci/config.yml",
        "jest.config.js",
        "appveyor.yml",
      ],
      "ignoreUpdates": Object {
        ".circleci/config.yml": false,
        "appveyor.yml": false,
        "jest.config.js": false,
        "wallaby.config.js": false,
      },
    },
    "makeDirs": Array [
      ".circleci",
      "dist",
      "src",
    ],
    "src": Object {
      "absolute": Object {
        "lib/files/appveyor.yml": "<PROJECT_ROOT>/lib/files/appveyor.yml",
        "lib/files/circleci-v2.yml": "<PROJECT_ROOT>/lib/files/circleci-v2.yml",
        "lib/files/jest-library.js": "<PROJECT_ROOT>/lib/files/jest-library.js",
        "lib/files/wallaby.js": "<PROJECT_ROOT>/lib/files/wallaby.js",
      },
      "files": Array [
        "lib/files/wallaby.js",
        "lib/files/circleci-v2.yml",
        "lib/files/jest-library.js",
        "lib/files/appveyor.yml",
      ],
      "hash": Object {
        "lib/files/appveyor.yml": "e26fc69a9c740e6fe6267ea2da435aa54c83a110eface66642470dc2e8336713",
        "lib/files/circleci-v2.yml": "3b7696d7b955ebd4ac0f5eeacfff227acea15301b6e450b111677d32411b2a8b",
        "lib/files/jest-library.js": "322dd136d43514064c38fdf8cd999d6511d83af8efbb35d8d355a3f35937c6c4",
        "lib/files/wallaby.js": "bfd3a73af1c8ecd93d812d70e4e599406f14eda36920955df69a4109a3a2ed2e",
      },
    },
  },
  "packageJson": Array [
    Object {
      "scripts": Object {
        "test.update": "jest --updateSnapshot",
        "test.watch": "jest --watch",
      },
    },
  ],
  "prepush": Array [
    "backtrack test",
  ],
  "resolve": Object {
    "@backtrack/preset-jest": "<PROJECT_ROOT>/lib",
  },
  "test": Array [
    "jest --passWithNoTests",
  ],
  "test.ci": Array [
    "jest --passWithNoTests --coverage --maxWorkers 2",
    Object {
      "name": "codecov",
      "task": "codecov",
    },
  ],
  "test.ci-pretest": Array [
    [Function noop],
  ],
}
`;

exports[`preset returns expected config - isApp = true, windows false 1`] = `
Object {
  "files": Object {
    "dest": Object {
      "absolute": Object {
        ".circleci/config.yml": "<PROJECT_ROOT>/.circleci/config.yml",
        "jest.config.js": "<PROJECT_ROOT>/jest.config.js",
        "wallaby.config.js": "<PROJECT_ROOT>/wallaby.config.js",
      },
      "allowChanges": Object {
        ".circleci/config.yml": false,
        "jest.config.js": false,
        "wallaby.config.js": false,
      },
      "files": Array [
        "wallaby.config.js",
        ".circleci/config.yml",
        "jest.config.js",
      ],
      "ignoreUpdates": Object {
        ".circleci/config.yml": false,
        "jest.config.js": false,
        "wallaby.config.js": false,
      },
    },
    "makeDirs": Array [
      ".circleci",
      "dist",
      "src",
    ],
    "src": Object {
      "absolute": Object {
        "lib/files/circleci-v2.yml": "<PROJECT_ROOT>/lib/files/circleci-v2.yml",
        "lib/files/jest-app.js": "<PROJECT_ROOT>/lib/files/jest-app.js",
        "lib/files/wallaby.js": "<PROJECT_ROOT>/lib/files/wallaby.js",
      },
      "files": Array [
        "lib/files/wallaby.js",
        "lib/files/circleci-v2.yml",
        "lib/files/jest-app.js",
      ],
      "hash": Object {
        "lib/files/circleci-v2.yml": "3b7696d7b955ebd4ac0f5eeacfff227acea15301b6e450b111677d32411b2a8b",
        "lib/files/jest-app.js": "4a1c02ec2ad2ed327246fea56e516a92bd7b57fedaba14da38faaa117410670d",
        "lib/files/wallaby.js": "bfd3a73af1c8ecd93d812d70e4e599406f14eda36920955df69a4109a3a2ed2e",
      },
    },
  },
  "packageJson": Array [
    Object {
      "scripts": Object {
        "test.update": "jest --updateSnapshot",
        "test.watch": "jest --watch",
      },
    },
  ],
  "prepush": Array [
    "backtrack test",
  ],
  "resolve": Object {
    "@backtrack/preset-jest": "<PROJECT_ROOT>/lib",
  },
  "test": Array [
    "jest --passWithNoTests",
  ],
  "test.ci": Array [
    "jest --passWithNoTests --coverage --maxWorkers 2",
    Object {
      "name": "codecov",
      "task": "codecov",
    },
  ],
  "test.ci-pretest": Array [
    [Function noop],
  ],
}
`;

exports[`preset returns expected config - windows = false 1`] = `
Object {
  "files": Object {
    "dest": Object {
      "absolute": Object {
        ".circleci/config.yml": "<PROJECT_ROOT>/.circleci/config.yml",
        "jest.config.js": "<PROJECT_ROOT>/jest.config.js",
        "wallaby.config.js": "<PROJECT_ROOT>/wallaby.config.js",
      },
      "allowChanges": Object {
        ".circleci/config.yml": false,
        "jest.config.js": false,
        "wallaby.config.js": false,
      },
      "files": Array [
        "wallaby.config.js",
        ".circleci/config.yml",
        "jest.config.js",
      ],
      "ignoreUpdates": Object {
        ".circleci/config.yml": false,
        "jest.config.js": false,
        "wallaby.config.js": false,
      },
    },
    "makeDirs": Array [
      ".circleci",
      "dist",
      "src",
    ],
    "src": Object {
      "absolute": Object {
        "lib/files/circleci-v2.yml": "<PROJECT_ROOT>/lib/files/circleci-v2.yml",
        "lib/files/jest-app.js": "<PROJECT_ROOT>/lib/files/jest-app.js",
        "lib/files/wallaby.js": "<PROJECT_ROOT>/lib/files/wallaby.js",
      },
      "files": Array [
        "lib/files/wallaby.js",
        "lib/files/circleci-v2.yml",
        "lib/files/jest-app.js",
      ],
      "hash": Object {
        "lib/files/circleci-v2.yml": "3b7696d7b955ebd4ac0f5eeacfff227acea15301b6e450b111677d32411b2a8b",
        "lib/files/jest-app.js": "4a1c02ec2ad2ed327246fea56e516a92bd7b57fedaba14da38faaa117410670d",
        "lib/files/wallaby.js": "bfd3a73af1c8ecd93d812d70e4e599406f14eda36920955df69a4109a3a2ed2e",
      },
    },
  },
  "packageJson": Array [
    Object {
      "scripts": Object {
        "test.update": "jest --updateSnapshot",
        "test.watch": "jest --watch",
      },
    },
  ],
  "prepublishOnly": Array [
    "backtrack test",
  ],
  "prepush": Array [
    "backtrack test",
  ],
  "resolve": Object {
    "@backtrack/preset-jest": "<PROJECT_ROOT>/lib",
  },
  "test": Array [
    "jest --passWithNoTests",
  ],
  "test.ci": Array [
    "jest --passWithNoTests --coverage --maxWorkers 2",
    Object {
      "name": "codecov",
      "task": "codecov",
    },
  ],
  "test.ci-pretest": Array [
    [Function noop],
  ],
}
`;
